### PR TITLE
feat(Modularization): init split sdk GCM, FCM, Push.

### PR DESCRIPTION
**Pull request by Enuviel**

### CODE REVIEW

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

File: AndroidSDKCore/src/main/java/com/leanplum/messagetemplates/MessageTemplates.java [hunk 1]
```diff
@@ -89,18 +89,18 @@
...
-    static final String DEFAULT_OPEN_URL = "http://leanplum:loadFinished";
-    static final String DEFAULT_TRACK_URL = "http://leanplum:track";
-    static final String DEFAULT_ACTION_URL = "http://leanplum:runAction";
-    static final String DEFAULT_TRACK_ACTION_URL = "http://leanplum:runTrackedAction";
+    static final String DEFAULT_OPEN_URL = DEFAULT_BASE_URL + "loadFinished";
+    static final String DEFAULT_TRACK_URL = DEFAULT_BASE_URL + "track";
+    static final String DEFAULT_ACTION_URL = DEFAULT_BASE_URL + "runAction";
+    static final String DEFAULT_TRACK_ACTION_URL = DEFAULT_BASE_URL + "runTrackedAction";
 
   }
```
Comment (acconway) [inline AndroidSDKCore/src/main/java/com/leanplum/messagetemplates/MessageTemplates.java]: Why are these changes a part of this commit?

Comment (Enuviel) [reply AndroidSDKCore/src/main/java/com/leanplum/messagetemplates/MessageTemplates.java]: they were merged from PR to this branch. Then we will merge this one to develop.

File: AndroidSDKFcm/build.gradle [hunk 2]
```diff
@@ -0,0 +1,144 @@
+apply plugin: 'com.android.library'
+apply plugin: 'com.jfrog.artifactory'
+apply plugin: 'maven-publish'
+
+def COMPILE_SDK_VERSION=27
+def BUILD_TOOLS_VERSION='27.0.2'
```
Comment (acconway) [inline AndroidSDKFcm/build.gradle]: In all other files there are spaces on each side of `=`

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Comment by Enuviel** [comment]
test this please

**Pull request merged**